LaunchDarkly Java SDK 6.x & 7.x Migration

shanman190 opened the pull request:
## What's changed?
Perform necessary migrations for the LaunchDarkly Java SDK 5.x to 6.x

## What's your motivation?
Fixes gh-3

## Anyone you would like to review specifically?
@timtebeek 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've added the license header to any new files through `./gradlew licenseFormat`
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

shanman190 committed:
Migrate `LDUser` to `LDContext`

shanman190 commented on src/test/java/org/openrewrite/launchdarkly/UpgradeLaunchDarkly6Test.java:
```diff
@@ -92,44 +94,154 @@ void mavenDependency() {
+                                  .set("email", "sandy@example.com")
+                                  .privateAttributes("email")
+                                  .set("groups", LDValue.buildArray().add("Google").add("Microsoft").build())
+                                  .privateAttributes("groups")
```
The `privateAttributes(java.lang.String...)` accepts varargs, but I didn't spot an easy way without possibly using a secondary visitor to add the private attribute method with all of the arguments at once.

Ideal after:
```java
import com.launchdarkly.sdk.LDContext;
import com.launchdarkly.sdk.LDValue;
                  
class A {
    void foo() {
        LDContext user = LDContext.builder("user-key-123abc")
                .name("Sandy")
                .set("email", "sandy@example.com")
                .set("groups", LDValue.buildArray().add("Google").add("Microsoft").build())
                .privateAttributes("email", "groups")
                .build();
    }
}                                 
```

timtebeek commented on src/test/java/org/openrewrite/launchdarkly/UpgradeLaunchDarkly6Test.java:
```diff
@@ -39,13 +41,13 @@ public void defaults(RecipeSpec spec) {
             .scanRuntimeClasspath("org.openrewrite.launchdarkly")
             .build()
             .activateRecipes("org.openrewrite.launchdarkly.UpgradeLaunchDarkly6"))
```
These lines above can be simplified to
```
        spec.recipeFromResource("/META-INF/rewrite/launchdarkly-6.yml", "org.openrewrite.launchdarkly.UpgradeLaunchDarkly6")
```

timtebeek reviewed (approved):
Looking very good! Thanks for taking this on.

shanman190 committed:
Flatten private attributes using varargs

shanman190 committed:
Cleanup based on PR review

shanman190 committed:
Add 7.x migration

timtebeek committed:
Inline isApplicableMethod, as the matcher checks null & instanceof

timtebeek committed:
Collapse dependency recipes to have less at top level

Unlikely folks would run these separately, as it's only one dependency; if they would want to they can run `UpgradeDependencyVersion` separately.

timtebeek committed:
Add language hints for code highlighting in text blocks

timtebeek committed:
Drop unused launchdarkly-java-server-sdk-7.1.1.jar for now
